src/averageAges.js

```diff
@@ -9,17 +9,26 @@
  * To calculate century:
  * Divide year of person's death by 100: Math.ceil(person.died / 100)
  *
+ * 1. write code here
+ * 2. learn how to use array methods like .filter .map .some .every .find .reduce
+ * 3. avoid using loop and forEach
+ * 4. replace `if ()` statement with &&, || or ?:
+ * 5. without nesting
+ *
  * @param {object[]} people
  * @param {number} century - optional
  *
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const filteredPeople = century
+    ? people.filter(person => Math.ceil(person.died / 100) === century
+      && person.sex === 'm')
+    : people.filter(person => person.sex === 'm');
+
+  return filteredPeople
+    .map(person => person.died - person.born)
+    .reduce((sum, age) => sum + age, 0) / filteredPeople.length;
```

Comment:
let's create helper functions for 
 - filtering people by gender
 - calculating average age
and use them wherever we need